File: BackEnd/src/main/java/com/example/granjasa/Entity/Reporte.java
package com.example.granjasa.Entity;

import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = "reporte")
public class Reporte {
    
    @Id
    private Integer idReporte;
    public Cliente clientes;
    public List<Porcino> porcinos;
}

File: FrontEnd/GranjaSA/src/app/main/Domain/Reporte.ts
import { Cliente } from "./Cliente";
import { Porcino } from "./Porcino";

export class Reporte {
    cliente: Cliente
    porcinos: Porcino[]

    constructor(cliente: Cliente, porcinos: Porcino[]) {
        this.cliente = cliente;
        this.porcinos = porcinos;
    }
}